### PR TITLE
Fix flaky dashboard tab tests

### PR DESCRIPTION
**Pull request by qnkhuat**
the root cause for these flake is that the `dashboard/ordered-cards` are ordered by creation time and sometimes in can change order.

I switched to a more reliable method by using position ordering.

### CODE REVIEW

**Review by calherries — APPROVED** [review]

File: src/metabase/models/dashboard_card.clj [hunk 1]
```diff
@@ -345,6 +345,14 @@
              dashcards))
       dashcards)))
 
+(defn dashcard-comparator
+  "Comparator that determines which of two dashcards comes first in the layout order used for pulses.
+  This is the same order used on the frontend for the mobile layout. Orders cards left-to-right, then top-to-bottom"
+  [dashcard-1 dashcard-2]
+  (if-not (= (:row dashcard-1) (:row dashcard-2))
+    (compare (:row dashcard-1) (:row dashcard-2))
+    (compare (:col dashcard-1) (:col dashcard-2))))
```
Comment (tsmacdonald) [inline src/metabase/models/dashboard_card.clj:354]:
```suggestion
  [{row-1 :row col-1 :col} {row-2 :row col-2 :col}]
  (if (= row-1 row-2)
    (compare col-1 col-2)
    (compare row-1 row-2)
```

^there are several distinct suggestions here, each of which are a little silly, feel free to pick and choose as you see fit. Another one is I think `(if (not= ...` reads better than `(if-not (= ...`.

Comment (tsmacdonald) [reply src/metabase/models/dashboard_card.clj:354]: (Ah, I now see you didn't even write the original code in the first place.)

File: src/metabase/pulse.clj [hunk 2]
```diff
@@ -92,14 +92,6 @@
     (catch Throwable e
       (log/warn e (trs "Error running query for Card {0}" card-or-id)))))
 
-(defn- dashcard-comparator
-  "Comparator that determines which of two dashcards comes first in the layout order used for pulses.
-  This is the same order used on the frontend for the mobile layout. Orders cards left-to-right, then top-to-bottom"
-  [dashcard-1 dashcard-2]
-  (if-not (= (:row dashcard-1) (:row dashcard-2))
-    (compare (:row dashcard-1) (:row dashcard-2))
-    (compare (:col dashcard-1) (:col dashcard-2))))
```
Comment (tsmacdonald) [inline src/metabase/pulse.clj:101]: thanks for DRYing this up

File: test/metabase/api/dashboard_test.clj [hunk 3]
```diff
@@ -1128,6 +1130,11 @@
                      (set (map :name cards-in-coll)))
                   "Cards should have \"-- Duplicate\" appended"))))))))
 
+(defn- ordered-cards-by-position
+  "Returned dashcards for a dashboard ordred by its position instead of creation like [[dashboard/ordered-cards]] did."
```
[the remark this answers is not in the document]
Comment (tsmacdonald) [reply test/metabase/api/dashboard_test.clj]: s/`creation`/`creation time,`

**Review by tsmacdonald — APPROVED** [review]

**Pull request merged**